### DISCUSSION TITLE
Count participants depending on demographic data

### DISCUSSION THREAD

**Discussion by slussinator** [ctreffe/alfred#214] [Alfred Q&A] [answered]
I would like to study two different populations using the same experiment. I need the same number of participants out of each population. Ideally, alfred would use information specified by participants at the beginning of the experiment (e.g. status: student vs. employed) to decide whether data for the respective population is still needed (and then go on with the experiment) or to stop the experiment once the quota for this population is fulfilled. 
Is this possible? And if so, how?

**Comment by jobrachem** [comment, marked as answer]
You can use `al.SessionQuota` for this purpose. Here is a minimal experiment:

```python
import alfred3 as al

exp = al.Experiment()

@exp.member
class DemographicPage(al.Page):

    def on_exp_access(self):
        self += al.SingleChoiceBar(
            "Student", "Employed",
            name="job",
            toplab="Please enter your employment status"
        )
    
    def on_first_hide(self):
        if self.exp.values.get("job", None) == 1: # if student
            quota = al.SessionQuota(nslots=1, exp=self.exp, name="quota_student")
            quota.count()
        else: # if employed
            quota = al.SessionQuota(nslots=1, exp=self.exp, name="quota_employed")
            quota.count()


exp += al.Page(title="Second page", name="second_page")



if __name__ == "__main__":
    exp.run()
```

The `SessionQuota` is the basic building block also for the `ListRandomizer`, meaning it does all the counting (but non of the randomizing). The `SessionQuota` will abort new sessions when `SessionQuota.count()` is called, and the required number of slots are finished, or if there are enough active sessions, such that the required number of slots may get filled. It behaves like the ListRandomizer in this regard.
